admin/app/components/solidus_admin/ui/forms/switch/component.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class SolidusAdmin::UI::Forms::Switch::Component < SolidusAdmin::BaseComponent
+  SIZES = {
+    s: '
+      w-8 h-5
+
+      after:w-4 after:h-4
+      after:top-0.5 after:left-0.5
+
+      active:after:w-4
+      after:peer-checked:left-[1.875rem]
+    ',
+    m: '
+      w-10 h-6
+
+      after:w-5 after:h-5
+      after:top-0.5 after:left-0.5
+
+      active:after:w-5
+      after:peer-checked:left-[2.375rem]
+    ',
+  }.freeze
+
+  def initialize(id: SecureRandom.hex, size: :m, **attributes)
+    @id = id
+    @size = size
+    @attributes = attributes
+    @states = []
+    @states << "checked" if @attributes[:checked]
+    @states << "disabled" if @attributes[:disabled]
```

Comment:
If we collect any attribute in the initializer we should apply it to the element, otherwise we can specifically collect just `checked` and `disabled`. Of course is unclear if the collected attributes would be applied to the label or the input.

Reply:
Agreed! I'll change it to specifically collect `checked` and `disabled`.